### ISSUE TITLE
Add id_token to TokenResponse

### ISSUE THREAD

**Issue by xhanin** [myndocs/kotlin-oauth2-server#52]
**Is your feature request related to a problem? Please describe.**
It would be nice to be able to support OpenIDConnect ID Token in the TokenResponse issued by the authorize calls. 

**Describe the solution you'd like**
Related to #50 which would allow to add custom metadata to AT and open the `Identity` information, it would be nice to be able to create an ID token in the Token response to comply with OIDC specification

**Describe alternatives you've considered**
The TokenResponse being closed and the conversion from AT to TokenResponse hard coded I see no other way to handle that, except relying on a ThreadLocal :(

**Comment by adhesivee** [comment]
Please check #53 if this supports your use-case

**Comment by xhanin** [comment]
Having metadata on `Identity` would help, but I would also need to have a way to add an id token to the token response.

So to support OIDC you would need to either:
* add an `idToken` property on `TokenReponse` or make it extensible, and give a way to customize the `fun AccessToken.toTokenResponse` extension method. 
* add an `idToken:String` property to `AccessToken` and `TokenResponse` (copied from one to the other in `AccessToken.toTokenResponse`), so that one could fill it in the `AccessTokenConverter`.

**Comment by adhesivee** [comment]
Ah yeah, overlooked that part. I created #54 which will allow this. 
This will give you the option to configure an access token responder in your configuration:
```
accessTokenResponder = object : AccessTokenResponder {
	override fun createResponse(accessToken: AccessToken): Map<String, Any?> =
            with(accessToken) {
                mapOf(
			"id_token" to "<YOUR_ID_TOKEN>",
                        "access_token" to this.accessToken,
                        "token_type" to this.tokenType,
                        "expires_in" to this.expiresIn(),
                        "refresh_token" to this.refreshToken?.refreshToken
                )
        }
}
```

**Comment by xhanin** [comment]
Sounds great, if we identity replace username in AccessToken, Identity is extensible and you allow to customize the TokenResponse then it's perfect!

**Comment by adhesivee** [comment]
Fixed with `0.5.0`
